README.rst
.. image:: https://travis-ci.org/collective/collective.prettyphoto.png
   :target: https://travis-ci.org/collective/collective.prettyphoto

.. contents::

Introduction
============

prettyPhoto_ is a jQuery based lightbox clone. It supports images,
it also add support for videos, flash, YouTube, iFrames and ajax.

It's a full blown media lightbox. The setup is easy and quick, plus the
script is compatible in every major browser.

This plugin has been tested and is known to work in the following
browsers:

* Firefox 3.0+
* Google Chrome 10.0+
* Internet Explorer 6.0+
* Safari 3.1.1+
* Opera 9+


`collective.prettyphoto` integrates prettyPhoto 3.1.5 into Plone.

Installing
==========

Development focus is now set to Plone 4.x. Currently all 4.x versions
are tested on `Travis CI`_. Plone 3.x is no longer officially supported.


Installing without buildout
---------------------------

Install this package in either your system path packages or in the
``lib/python`` directory of your Zope instance. You can do this using
either easy_install or via the ``setup.py`` script.

Installing with buildout
------------------------

If you are using buildout_ to manage your instance installing
collective.prettyphoto is even simpler. You can install
collective.prettyphoto by adding it to the eggs line for your instance::

    [instance]
    eggs = collective.prettyphoto

After updating the configuration you need to run the ``bin/buildout``,
which will take care of updating your system.


Usage
=====

collective.prettyphoto adds a new view for Collections and Folders:
Thumbnail view (prettyPhoto).

To use prettyPhoto for inline elements just add ``prettyPhoto`` from the
styles menu in TinyMCE to the link.


Configuration
=============

collective.prettyphoto can be customized via property sheet (go to ZMI,
portal_properties, prettyphoto_properties).

* theme:

  * pp_default (default)

  * dark_rounded

  * dark_square

  * facebook

  * light_rounded

  * light_square

* speed:

  * fast

  * normal (default)

  * slow

* opacity: value from ``0.0`` to ``1.0``
  (default: ``0.80``)

* show_title: show the title for images?
  (default: ``True``)

* counter_sep: the separator for the gallery counter 1 "of" 2
  (default: "/")

* autoplay: automatically start videos?
  (default: ``True``)

* iframe_width: the width of the iframe
  (must be percantage, default: ``75%``)

* iframe_height: the height of the iframe
  (must be percantage, default: ``75%``)

* overlay_gallery: If set to true, a gallery will overlay the fullscreen image
  on mouse over (default: ``False``, the bug (http://bit.ly/eKYdrF) is fixed
  since 0.4.1 (pP 3.0.3) but your image url has to contain
  (jpg|jpeg|png|gif) somewhere)

* slideshow: interval time in ms, ``0`` to disable slideshow
  (default: ``0``)

* autoplay_slideshow: automatically start slideshow?
  (default: ``False``, only woks when slideshow > ``0``)

* deeplinking: rewrite the browser location to enable directlinks to
  overlay images (default: ``False``)

* social_tools: html markup for social links within image overlays
  (twitter, FB, ...) (default: empty) see http://bit.ly/pDQ19A for detailed
  information


Copyright and Credits
=====================

prettyPhoto_ is developed by Stephane Caron
(http://www.no-margin-for-errors.com) and is licensed under Creative
Commons Attribution 2.5.

Contributors (alphabetically)
-----------------------------

* fRiSi
* hpeteragitator
* jensens
* petschki
* saily
* tmassman

.. _prettyPhoto: http://www.no-margin-for-errors.com/projects/prettyphoto-jquery-lightbox-clone/
.. _buildout: http://pypi.python.org/pypi/zc._buildout
.. _`Travis CI`: http://travis-ci.org/collective/collective.prettyphoto


docs/HISTORY.rst
Changelog
=========

0.7 (unreleased)
----------------

- Nothing changed yet.


0.6 (2016-02-10)
----------------

- Fix zcml condition on Products.kupu
  [ale-rt]

0.5 (2013-04-18)
----------------

- Documentation updates
  [saily]

- Upgrade to prettyPhoto 3.1.5
  [saily]

- Add ui-tests using robotframework to validate views.
  [saily]

- Add travis integration for Plone 4.3.x, Plone 4.2.x, Plone 4.1.x and
  Plone 4.0.x. Plone 3.3.x cannot be tested with new layered testing structure
  of ``plone.app.testing`` and will never be testable on travis because they
  don't and will never support Python 2.4.

  So we changed primary focus of development to Plone >= 4.0.
  [saily]

- Switchted to ``plone.app.testing`` and added some basic installation tests.
  [saily]

- Add egg-containing buildout and bootstrap.py
  [saily]

- Use ``zcml:condition="installed plone.app.collection"`` to switch to a new
  GS profile which appends prettyphoto views to Collection instead Topic type.
  [saily]

- Refactor GS profiles and depend each profile on a 'extended' profile which
  hold generic stuff as registering css and js. Generic setup takes the first
  profile which is found in folder, so this has to be alpha-sorted behind
  'default'.
  [saily]


0.4.5.2 (2012-09-12)
--------------------
- more manifest packaging madness

0.4.5.1 (2012-09-12)
--------------------

- fix bug with sdist packaging
  [jensens]

0.4.5 (2012-09-11)
------------------

- corrected the global js-function
  [bennyboy]

- Made it possible to later (i.e. after ajax calls) bind prettyphoto
  to images. thus we have a global js-function plonePrettyPhotoEnable now.
  Also did some housekeeping, added integrated buildout, gitignore, ...
  [jensens]

0.4.4 (2011-12-02)
------------------

- Added possiblity to provide custom markup by subclassing the
  view that creates the prettyphoto configuration.
  (see ``browser.view.Javascript.markup``)
  [fRiSi]

0.4.3 (2011-07-22)
------------------

- Fixed parameter names used to configure prettyPhoto
  and added autoplay_slideshow.

  Previously titles have been shown although ``show_title`` has been set to
  ``False`` in ``prettyphoto_properties``.
  [fRiSi]

- Finish update to 3.1.2 by updating css and images and adding the new `pp_default`
  theme (which is the new default btw ;-)
  [fRiSi]

0.4.2 (2011-07-13)
------------------

- Upgraded to prettyPhoto 3.1.2 and implemented two new properties:
  deeplinking , social_tools
  [petschki]

0.4.1 (2011-03-22)
------------------

- Upgraded to prettyPhoto 3.0.3 .
  This fixes thumbnails in overlay_gallery partly because the regular expression
  now looks for a (jpg|jpeg|png|gif) within the whole url. if your originaly
  linked images are completely without this, they still wont show up.
  [petschki]

- fixed overlay_gallery property defaults to "False"
  [petschki]

0.4.0 (2011-02-23)
------------------

- Upgraded to prettyPhoto library including it's css and image sprite to 3.0.1
  to fix issues with jQuery 1.4.4 (see http://bit.ly/hxYUrt) which is shipped
  with latest Plone 4 by default. This fixes #2.
  [saily]

- Added default thumbnails for 'overlay_galleries' feature.
  [saily]

- Added 'overlay_gallery' and 'slideshow' as additional configuration options
  but disabled by default cause of a bug in jquery.prettyPhoto.js which avoids
  displaying thumbnails not ending on (jpg|jpeg|png|gif).
  See http://bit.ly/eKYdrF
  [saily]

0.3.3 (2010-12-20)
------------------

- Fixed iframe configuration (width was used for height too)
  [fRiSi]

0.3.2 (2010-07-10)
------------------

- Fixed install issue with Plone 4 (Large Plone Folder was removed).
  [hpeteragitator]

- Set version in metadata.xml to 1, since this has nothing to do with the package version.
  [tmassman]

0.3.1 (2010-05-19)
------------------

- Fixed version numbers and missing upgrade steps.
  [tmassman]

- Added custom browserlayer.
  [tmassman]

- Upgrade to prettyPhoto 2.5.6
  [tmassman]

0.3 (2010-05-17)
----------------

- Don't fail when installing on sites w/o kupu (plone4 compatibility).
  [fRiSi]

0.2 (2010-01-03)
----------------

- Fixed wrong kupu styles.
  [tmassman]

- Added iFrame support (use 'prettyPhoto Iframe Link' for external sites)
  [tmassman]


0.1 (2009-12-29)
----------------

- Initial release
  [tmassman]



collective/prettyphoto/browser/view.py
from Products.Five.browser import BrowserView
from Products.CMFCore.utils import getToolByName


class JavaScript(BrowserView):

    @property
    def prettyphoto_properties(self):
        properties_tool = getToolByName(self.context, 'portal_properties')
        return getattr(properties_tool, 'prettyphoto_properties', None)

    def __call__(self, request=None, response=None):
        """Returns global configuration for prettyPhoto taken from
           portal_properties."""
        self.request.response.setHeader("Content-type", "text/javascript")

        social_tools = getattr(self.prettyphoto_properties, 'social_tools', '')


        slideshowInterval = getattr(self.prettyphoto_properties, 'slideshow', 0)
        autoplaySlideshow = getattr(self.prettyphoto_properties, 'autoplay_slideshow', False)
        if slideshowInterval <= 0:
            # automatically starting the slideshow does only make sense if the interval is greater than 0
            autoplaySlideshow = False

        return """\
        var plonePrettyPhoto = {};
        (function($) {
            plonePrettyPhoto.enable = function () {
                // add rel tag for all links with class 'prettyPhoto'
                $("a.prettyPhoto").attr({"rel": "prettyPhoto"});

                // add iframe attributes for all links with class 'prettyPhotoIframe'
                $("a.prettyPhotoIframe").attr("href", function() {
                    return this.href + "?iframe=true&width=%(iframe_width)s&height=%(iframe_height)s";
                }).attr({"rel": "prettyPhoto"});

                // enable prettyPhoto
                $("a[rel^='prettyPhoto']").prettyPhoto({
                    animation_speed: '%(speed)s', /* fast/slow/normal */
                    opacity: %(opacity)s, /* Value between 0 and 1 */
                    show_title: %(show_title)s, /* true/false */
                    counter_separator_label: '%(counter_sep)s', /* The separator for the gallery counter 1 "of" 2 */
                    theme: '%(theme)s',
                    autoplay: %(autoplay)s, /* Automatically start videos: true/false */
                    autoplay_slideshow: %(autoplay_slideshow)s,
                    slideshow: %(slideshow)s, /* false OR interval time in ms */
                    overlay_gallery: %(overlay_gallery)s, /* If set to true, a gallery will overlay the fullscreen image on mouse over */
                    social_tools: %(social_tools)s, /* html markup for social tool icons */
                    deeplinking: %(deeplinking)s, /* allow prettyphoto to rewrite url for direktlinking to an image */
                    markup: %(markup)s
                });
            };

            $(function() {
                // enable prettyPhoto on document load
                plonePrettyPhoto.enable();
            });
        })(jQuery);
        """ % dict(speed=getattr(self.prettyphoto_properties, 'speed', 'normal'),
                   opacity=getattr(self.prettyphoto_properties, 'opacity', '0.80'),
                   show_title=getattr(self.prettyphoto_properties, 'show_title', True) and 'true' or 'false',
                   counter_sep=getattr(self.prettyphoto_properties, 'counter_sep', '/'),
                   theme=getattr(self.prettyphoto_properties, 'theme', 'light_rounded'),
                   autoplay=getattr(self.prettyphoto_properties, 'autoplay', True) and 'true' or 'false',
                   iframe_width=getattr(self.prettyphoto_properties, 'iframe_width', '75%'),
                   iframe_height=getattr(self.prettyphoto_properties, 'iframe_height', '75%'),
                   overlay_gallery=getattr(self.prettyphoto_properties, 'overlay_gallery', False) and 'true' or 'false',
                   slideshow=slideshowInterval or 'false',
                   autoplay_slideshow=autoplaySlideshow  and 'true' or 'false',
                   social_tools=social_tools and "'%s'" % social_tools or 'false',
                   deeplinking=getattr(self.prettyphoto_properties, 'deeplinking', False) and 'true' or 'false',
                   markup=self.markup() or 'undefined',
              )

    def markup(self):
        """subclasses can overwrite this to provide another default markup
        for the overlay.

        eg:
        return '''"<div class="pp_pic_holder"> \\
            <div class="pp_top"> \\
            ..." '''
        """
        return None
